src/Helpers/DataResolver.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+namespace Firesphere\SolrSearch\Helpers;
+
+use Exception;
+use SilverStripe\Core\ClassInfo;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\FieldType\DBField;
+use SilverStripe\ORM\SS_List;
+use SilverStripe\View\ArrayData;
+
+class DataResolver
+{
+    protected $component;
+    protected $columns = [];
+    protected $columnName = '';
+
+    public function __construct($component, $columns = [])
+    {
+        if (!is_array($columns)) {
+            $columns = array_filter(explode('.', $columns));
+        }
+        $this->columns = $columns;
+        $this->component = $component;
+        $this->columnName = $this->columns ? array_shift($this->columns) : null;
+    }
+
+    /**
+     * @param DataObject|ArrayData|SS_List $obj
+     * @param array|string $columns
+     *
+     * @return mixed
+     * @throws Exception
+     */
+    public static function identify($obj, $columns = [])
+    {
+        $instance = new self($obj, $columns);
+        if ($obj instanceof DataObject) {
+            return $instance->resolveDataObject();
+        }
+        if ($obj instanceof ArrayData) {
+            return $instance->resolveArrayData();
+        }
+        if ($obj instanceof SS_List) {
+            return $instance->resolveList();
+        }
+        if ($obj instanceof DBField) {
+            return $instance->resolveField();
+        }
+        throw new Exception(sprintf('Class: %s, is not supported.', ClassInfo::shortName($obj)));
+    }
+
+    /**
+     * @param DataObject|ArrayData|SS_List $component
+     * @param array $columns
+     *
+     * @return void
+     * @throws Exception
+     */
+    protected function cannotIdentifyException($component, $columns = [])
+    {
+        throw new Exception(
+            sprintf(
+                'Cannot identify, "%s" from class "%s"',
+                implode('.', $columns),
+                ClassInfo::shortName($component)
+            )
+        );
+    }
+
+    /**
+     * Resolves a Single field in the database.
+     * @return mixed
+     * @throws Exception
+     */
+    protected function resolveField()
+    {
+        if ($this->columnName) {
+            $method = $this->columnName;
+            if (!method_exists($this->component, $method)) {
```

Comment:
Method could also be `get{$method}`, something to check too. The framework will return true for either if either exists, but return false if both the direct and indirect method exist.

This is to ensure backward compatibility with FTS.

It could be a bit more around in this commit, where this is applicable, but this one rang a bell :)

Reply:
This is great! 👍